ui/webui/resources/br_elements/br_toolbar/br_toolbar.ts

```diff
@@ -1,167 +1,216 @@
+      showMenu: { type: Boolean },
+
+      // Whether to show menu promo tooltip.
+      showMenuPromo: { type: Boolean },
+
+      // Controls whether the search field is shown.
+      showSearch: { type: Boolean },
+
+      // True when the toolbar is displaying in narrow mode.
+      narrow: { type: Boolean, reflect: true },
+
+      /**
+       * The threshold at which the toolbar will change from normal to narrow
+       * mode, in px.
+       */
+      narrowThreshold: { type: Number },
+
+      closeMenuPromo: { type: String },
+
+      noSearch: { type: Boolean },
+
+      /** @private */
+      showingSearch_: {
+        type: Boolean,
+        reflect: true,
+      },
+
+      shouldShowRewardsButton_: {
+        type: Boolean,
+      },
+
+      isBraveWalletAllowed_: {
+        type: Boolean,
+      },
+
+      fontsLoadedClassName: {
+        type: String
+      }
+    }
+  }
+
+  pageName = ''
+  searchPrompt = ''
+  clearLabel = ''
+  menuLabel = ''
+  menuPromo = ''
+  spinnerActive = false
+  showMenu = false
+  showMenuPromo = false
+  showSearch = true
+  narrow = false
+  narrowThreshold = 900
+  narrowQuery_: MediaQueryList | null = null
+  closeMenuPromo = ''
+  noSearch = false
+  showingSearch = false
+  showRewardsButton = true
+  isBraveWalletAllowed_ = loadTimeData.getBoolean('brToolbarShowRewardsButton')
+
+  // Localized strings:
+  historyTitle = loadTimeData.getString('brToolbarHistoryTitle')
+  settingsTitle = loadTimeData.getString('brToolbarSettingsTitle')
+  bookmarksTitle = loadTimeData.getString('brToolbarBookmarksTitle')
+  downloadsTitle = loadTimeData.getString('brToolbarDownloadsTitle')
+  braveRewardsTitle = loadTimeData.getString('brToolbarRewardsTitle')
+  walletsTitle = loadTimeData.getString('brToolbarWalletsTitle')
+
+  // Settings from `loadTimeData`
+  shouldShowRewardsButton_ = loadTimeData.getBoolean('brToolbarShowRewardsButton')
+
+  // Non-observed properties
+  fontsLoadedClassName = ''
+
+  // Slotted content
+  toolbarExtraSlot: HTMLSlotElement
+  toolbarExtraElement: Element
+
+  override updated(changedProperties: PropertyValues<this>) {
+    super.updated(changedProperties);
+    if (changedProperties.has('narrowThreshold')) {
+      this.narrowQuery_ = window.matchMedia(`(max-width: ${this.narrowThreshold}px)`);
+      this.narrow = this.narrowQuery_.matches
+      this.narrowQuery_.addEventListener('change', () => {
+        this.narrow = this.narrowQuery_?.matches ?? false;
+      });
+    }
+
+    if (changedProperties.has('noSearch')) {
+      this.noSearchChanged_()
+    }
+  }
 
   /** @return {!CrToolbarSearchFieldElement} */
-  getSearchField: function() {
+  getSearchField() {
     return this.$.search;
-  },
+  }
 
   /** @private */
-  onMenuTap_: function() {
+  onMenuClick_() {
     this.dispatchEvent(new CustomEvent(
-      'cr-toolbar-menu-click', {bubbles: true, composed: true}))
-  },
+      'cr-toolbar-menu-click', { bubbles: true, composed: true }))
```

Comment:
How about `@click=${() => this.fire('cr-toolbar-menu-click')`?

Reply:
Hey, that's super cool, I didn't know about it. So I'm going to use `this.fire` but leave in this `onMenuClick_` function because it's marked as `protected` in Chromium and we need to try and match their API :laughing: